NewForce-at-Mountwest/bangazon-api-farfalle
# Controller for computer resource supports get, post, put and delete.  Test mechanism included

# Description

I have created a controller for the payment type resource that supports get all, get single, post, put and delete (soft delete and hard delete).  I added the API Client provider to the TestBangazonAPI folder for all future test applications


## Related Ticket(s)
Ticket [8](https://trello.com/c/KWtsDSkZ/8-ticket-8-allow-developers-to-access-the-computers-resource)


## Steps to Test

Be sure to add and commit on your own branch before checking out to master!
```git fetch --all``` to get access to my branch

```git checkout sw-computer-controller ```
```git pull origin sw-computer-controller```

type ```start BangazonAPI.sln``` if you don't already have it open

In the Solution Explorer, right click on  'TestBangazonApi' and select 'Run Tests'

Because this controller only 'soft deletes' things from the database by setting the DecomissionDate to a non-null value, I had to hard-code hard-delete functionality.  To ensure that it ACTUALLY hard-deleted, you can use either SSMS or the sql viewer in Visual Code to perform a sql query on the BangazonAPI database:

SELECT * FROM computer

You can also check all functionality in Postman and you will be able to see a decommissioned computer by going to ````https://localhost:5001/api/computer/{id}````.  

To hard delete, you would use the url:
````https://localhost:5001/api/computer/8?HardDelete=true````
with a delete method.

You should NOT see anything that says "test" or "cool" in computers.  You should only see the computers already seeded in your database (we started with 3).  

Null Decommission Dates should show up in the browser and in sql as 'null' but they will show as "0001-01-01T00:00:00" in Postman.


# Checklist:
